test/gurka/test_access.cc

```diff
@@ -0,0 +1,621 @@
+
+  const gurka::nodes nodes = {{"F", {{"motor_vehicle", "no"}}}, {"G", {{"motorcar", "no"}}}};
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  auto map = gurka::buildtiles(layout, ways, nodes, {}, "test/data/gurka_node_access");
+
+  for (auto& c : costing) {
+    if (c == "auto" || c == "taxi")
+      validate_path(gurka::do_action(valhalla::Options::route, map, {"A", "I"}, c),
+                    {"AB", "BC", "CD", "DE", "EH", "HL", "KL", "JK", "IJ"});
+    else if (c == "bicycle" || c == "pedestrian")
+      validate_path(gurka::do_action(valhalla::Options::route, map, {"A", "I"}, c),
+                    {"AB", "BF", "FI"});
+    else
+      validate_path(gurka::do_action(valhalla::Options::route, map, {"A", "I"}, c),
+                    {"AB", "BC", "CD", "DG", "GK", "JK", "IJ"});
+  }
+}
+
+TEST(Standalone, RouteOnPrivateAccess) {
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+        A---B---C---D
+            |   |   |
+            E   F   G
+    )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "primary"}}},
+      {"BC", {{"highway", "primary"}}},
+      {"CD", {{"highway", "primary"}}},
+      {"BE", {{"highway", "service"}, {"access", "private"}}},
+      {"CF", {{"highway", "service"}, {"access", "private"}, {"service", "driveway"}}},
+      {"DG", {{"highway", "service"}, {"access", "private"}, {"service", "parking_aisle"}}},
+  };
+
+  const auto layout =
+      gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_route_on_private_access",
+                               build_config);
+
+  for (auto& c : costing) {
+    auto result = gurka::do_action(valhalla::Options::route, map, {"A", "E"}, c);
+    gurka::assert::raw::expect_path(result, {"AB", "BE"});
+
+    result = gurka::do_action(valhalla::Options::route, map, {"A", "F"}, c);
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CF"});
+
+    result = gurka::do_action(valhalla::Options::route, map, {"A", "G"}, c);
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DG"});
+  }
+}
+
+
+TEST(Standalone, AccessForwardBackward) {
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+          A--B--C-D-E
+                |   |
+                F   |
+                |   |
+                H---G
+    )";
+
+  const gurka::ways ways = {
+      {"ABCDE", {{"highway", "primary"}}},
+      {"CFH", {{"highway", "primary"}, {"motor_vehicle:forward", "no"}, {"motor_vehicle:backward", "yes"}, {"foot:forward", "no"}, {"foot:backward", "yes"}, {"bicycle:forward", "no"}, {"bicycle:backward", "yes"}}},
+      {"HG", {{"highway", "primary"}}},
+      {"EG", {{"highway", "primary"}, {"motor_vehicle:forward", "yes"}, {"motor_vehicle:backward", "no"}, {"foot:forward", "yes"}, {"foot:backward", "no"}, {"bicycle:forward", "yes"}, {"bicycle:backward", "no"}}},
+  };
```

Comment:
so EG can only be traversed from E to G for all modes and CFH can only be traversed from H to C by all modes